Add support for org-reveal to org layer

knl opened the pull request:
org-reveal is an extension to org mode that allows exporting of org files
as `reveal.js` presentations. Since not everyone needs this, the loading of
the package is controlled via `org-enable-reveal-js` switch, which, if set
to `t` would load the package.

robbyoconnor commented:
Rebase -- this should be one commit not 4.  Also see #2501

robbyoconnor commented:
And no submodules :|

knl commented:
Hi @robbyoconnor I can't rebase because git subtrees can't be squashed. Also, no submodules here, as the commit says - only subtrees :)

Now, the actual question is: should I include reveal.js or not. If not, I will need to add text for the user on how to download/set reveal.js location. If yes, then this has to remain as 4 commits (I can reduce it to 3, tho).

robbyoconnor commented:
This adds so much bloat -- I'd much rather just have the elisp stuff and let the end user grab reveal.js

knl commented:
Sure, that is doable. Would that be an official project's stance?

robbyoconnor commented:
With all that stuff removed -- you should be able to have one single, nice sexy commit -- and no merge conflicts!

robbyoconnor commented:
<3 <3

robbyoconnor commented:
Although, you should probably add a note about where they get reveal.js; and where ox-reveal expects it to be ;)

knl commented:
Thanks for the help and guidance @robbyoconnor, I updated the PR text and the patch.

robbyoconnor commented:
We *DO* need reveal.js -- but I don't feel like we need it in the tree (it increases bloat) -- so you need to tell users it's necessary ;)

robbyoconnor commented:
:+1: -- just re-read it :)

robbyoconnor commented on layers/org/README.org:
```diff
@@ -59,6 +60,21 @@ at this stage.  It is possible to disable the configuration of gnuplot support
+
+To enable the export of your org file as [[http://lab.hakim.se/reveal-js/][reveal.js]] presentation, set the
+variable =org-enable-reveal-js= to =t=. This will install [[https://github.com/yjwen/org-reveal/][org-reveal]] extension.
+
```
This is broken english.

robbyoconnor replied:
"To enable the export of your org file as [[http://lab.hakim.se/reveal-js/][reveal.js]] a presentation, set the variable =org-enable-reveal-js= to =t=. This will install [[https://github.com/yjwen/org-reveal/][org-reveal]] extension."

robbyoconnor replied:
"To enable the export of your org file as a [[http://lab.hakim.se/reveal-js/][reveal.js]] presentation, set the variable =org-enable-reveal-js= to =t=. This will install the [[https://github.com/yjwen/org-reveal/][org-reveal]] extension."

robbyoconnor commented on layers/org/README.org:
```diff
@@ -59,6 +60,21 @@ at this stage.  It is possible to disable the configuration of gnuplot support
+In order to be able to use org-reveal, download =reveal.js= from its 
+[[http://lab.hakim.se/reveal-js/#/][homepage]] and set =org-reveal-js= to the 
+download path, as described in the [[https://github.com/yjwen/org-reveal#obtain-revealjs][manual]].
+
```
:+1:

knl commented:
Thanks @robbyoconnor, the modifications are in. One could easily spot my slavic origins - we don't have any articles in our language :)

robbyoconnor commented:
It is completely okay :heart:

robbyoconnor commented:
Clean up merge conflicts

knl commented:
Done, the docs have been fixed.

robbyoconnor commented:
Can this be merged @syl20bnr, @TheBB ?

cpaulik commented:
Maybe we could add to the README that it is not really necessary to download reveal.js. You can just set `#+REVEAL_ROOT: http://cdn.jsdelivr.net/reveal.js/3.0.0/` in the org file. I know that this is mentioned in the original documentation. But it is rather buried for what I think is a very obvious use case.

knl commented:
Hi @cpaulik you're right, that is also a possibility, but I encountered some problems with exports, so I went for the bullet-proof option. I've incorporated your suggestion.

opyate commented:
Every time I upgrade Spacemacs, I loose this feature, and get it back with:

    cd ~/.emacs.d
    git remote add reveal https://github.com/knl/spacemacs
    git fetch reveal
    git checkout reveal/reveal.js

:+1: for a merge.

robbyoconnor commented:
Just fetch this on your branch and then rebase on top of it. No loss ;)

malchmih commented:
Would be great to have this merged! :+1:

nrolland commented:
what's needed to merge this ?

StreakyCobra commented:
@nrolland There are 158 other PRs waiting. :+1: the PR to show your interest.

robbyoconnor commented:
Probably resolve the merge  conflicts for one.

nrolland commented:
:)) the ransom of success !

kepi commented:
:+1:

knl commented:
I've updated the branch, merges well. Please let me know if someone needs some additional functionality.

Dietr1ch commented:
I've just used this for a presentation. It worked fine.

The only caveat is that the `org-enable-reveal-js` variable did not worked for me and I needed to manualy load the `ox-reveal` package before exporting (this may be a misunderstanding of mine, but anyways loading it should not add too much overhead).

robbyoconnor commented:
@Dietr1ch worked fine for me -- I used this.

knl commented:
Hi Dietr1ch, thanks for reporting the issue. I've updated the code to match the style in other `use-package` calls, and added `:init` part that `require`s the package. Confirmed to work, as previous one didn't correctly load the package.

Dietr1ch commented:
@knl That made it. I just merged and it worked out of the box (=

This seems ready for a merge.

syl20bnr commented:
Thank you ! 👍 
I renamed the layer variable to `org-enable-reveal-js-support` to be more like the other variables of the layer.
Cherry-picked into develop branch, you can safely delete your branch.
